tests/test_graph.py

```diff
@@ -113,27 +116,58 @@ def test_cycle() -> None:
 @pytest.mark.parametrize(
     ("mean", "stdev", "samples", "rtol"),
     [
-        pytest.param(1.0, 1.0, 10, 1, id="std normal, 10 samples"),
-        pytest.param(2.0, 0.8, 1000, 1e-1, id="non-standard normal, 100 samples"),
-        pytest.param(1.0, 1.0, 100000, 1e-2, id="std normal, 10^5 samples"),
-        pytest.param(1.0, 1.0, 10000000, 1e-3, id="std normal, 10^7 samples"),
+        pytest.param(1.0, 1.0, 10, 1, id="N(mean=1, stdev=1), 10 samples"),
+        pytest.param(2.0, 0.8, 1000, 1e-1, id="N(mean=2, stdev=0.8), 1000 samples"),
+        pytest.param(1.0, 0.8, 100000, 1e-2, id="N(mean=1, stdev=0.8), 10^5 samples"),
+        pytest.param(1.0, 1.2, 10000000, 1e-3, id="N(mean=1, stdev=1.2), 10^7 samples"),
     ],
 )
-def test_single_normal_node(samples, rtol, mean, stdev):
-    normal = causalprog.graph.node.NormalDistribution(mean, stdev)
-    node = causalprog.graph.DistributionNode(normal, "X", is_outcome=True)
+def test_single_normal_node(samples, rtol, mean, stdev, rng_key):
+    node = DistributionNode(
+        NormalFamily(),
+        "X",
+        constant_parameters={"mean": mean, "cov": stdev**2},
+        is_outcome=True,
+    )
 
-    graph = causalprog.graph.Graph("G0")
+    graph = Graph("G0")
     graph.add_node(node)
 
+    key = jax.random.split(rng_key, 1)[0]
```

Comment:
```suggestion
    # To compensate for rng-key splitting in sample methods, note the "split" key
    # that is actually used to draw the samples from the distribution, so we can
    # attempt to replicate its behaviour explicitly.
    key = jax.random.split(rng_key, 1)[0]
```

Just adding this comment here as between yesterday morning and now I had forgotten why we did this :sweat_smile: